src/script/lua_api/l_server.cpp

```diff
@@ -505,6 +506,191 @@ int ModApiServer::l_set_last_run_mod(lua_State *L)
 	return 0;
 }
 
+// common start: ensure auth db or push nil to Lua
+AuthDatabase *ModApiServer::getAuthDb(lua_State *L)
+{
+	ServerEnvironment *server_environment =
+			dynamic_cast<ServerEnvironment *>(getEnv(L));
+	if (!server_environment) {
+		lua_pushnil(L); // not ready yet
+		return nullptr;
+	}
+	AuthDatabase *auth_db = server_environment->getAuthDatabase();
+	if (!auth_db) {
+		lua_pushnil(L); // no database (yet)
```

Comment:
This is very inconsistent. On failure you're pushing nil to Lua, but not on success. I recommend to only return the pointers and push nil in the parent function.

Reply:
Yeah, it's very unclean; an attempt to share as much code as possible. I'll change this.

Reply:
Changed, also including the `return 0;` from your next review comment.